### PR TITLE
[NGINX] add extra location + htaccess property

### PR DESCRIPTION
**Pull request by vrenaville**
* Purpose
Some of our customers want to have access to a specific location ex : /special_api with a basic authentification
(htaccess).

### CODE REVIEW

File: 9.0/templates/nginx.conf.tmpl [hunk 1]
```diff
@@ -109,6 +109,24 @@ http {
...
+      {{ $odoo_extra_location_htaccess := getenv "NGX_ODOO_EXTRA_LOCATION_HTACCESS" }}
+      auth_basic_user_file $odoo_extra_location_htaccess;
+      {{ end }}
+      add_header X-Static no;
+      proxy_buffering off;
+      proxy_buffer_size 64k;
+      proxy_busy_buffers_size 64k;
+      proxy_intercept_errors on;
+    }
+   {{ end }}
```
Comment (gurneyalex) [inline 9.0/templates/nginx.conf.tmpl:129]: can you update the documentation too, with an example and the commands to generate the auth basic user file ?

Comment (vrenaville) [reply 9.0/templates/nginx.conf.tmpl:129]: @gurneyalex, Thanks for the review
sure, I just want to be sure of the approach first, before doing an entry in the README

**Comment by vrenaville** [comment]
TODO: https://github.com/kelseyhightower/confd/blob/master/docs/templates.md#advanced-map-traversals

File: 9.0/templates/nginx.conf.tmpl [hunk 2]
```diff
@@ -109,6 +109,26 @@ http {
...
+   {{ if getenv "NGX_ODOO_EXTRA_LOCATION" }}
+    {{ $odoo_extra_location := getenv "NGX_ODOO_EXTRA_LOCATION" }}
+    {{range jsonArray $odoo_extra_location}}
+    location {{.}} {
+      proxy_pass http://{{ $odoo_host }}:8069;
+      proxy_read_timeout    6h;
+      proxy_connect_timeout 5s;
+      proxy_redirect        off;
+      {{ if getenv "NGX_HTPASSWD" }}
+      auth_basic "Odoo exta location restricted";
```
Comment (guewen) [inline 9.0/templates/nginx.conf.tmpl:121]:
```suggestion
      auth_basic "Odoo extra location restricted";
```

**Review by guewen — APPROVED** [review]
only a small typo spotted

**Review by p-tombez — APPROVED** [review]
LGTM

**Pull request merged**